components/shared/searchFilters/SearchFiltersColumn.tsx

```diff
@@ -0,0 +1,90 @@
+import { SearchFacetsQuery } from "@/lib/graphql/generated/fbi/graphql"
+import { cn } from "@/lib/utils"
+import React, { useEffect, useRef, useState } from "react"
+import { mapFacetsToFilters, mapFilterNameToTranslation, toggleFilter } from "./helper"
+import { useRouter, useSearchParams } from "next/navigation"
+import Icon from "../icon/Icon"
+import { Button } from "../button/Button"
+
+type SearchFiltersColumnProps = {
+  facet: SearchFacetsQuery["search"]["facets"][0]
+  isLast: boolean
+  isExpanded: boolean
+  setIsExpanded: React.Dispatch<React.SetStateAction<boolean>>
+}
+
+const SearchFiltersColumn = ({
+  facet,
+  isLast,
+  isExpanded,
+  setIsExpanded,
+}: SearchFiltersColumnProps) => {
+  const router = useRouter()
+  const facetName = facet.name as keyof typeof mapFacetsToFilters
+  const searchParams = useSearchParams()
+  const elementRef = useRef<HTMLDivElement | null>(null)
+  const [hasOverflow, setHasOverflow] = useState(false)
+
+  useEffect(() => {
+    const el = elementRef.current
+    if (el) {
+      const isOverflowing = el.scrollHeight > el.clientHeight || el.scrollWidth > el.clientWidth
+      isOverflowing ? setHasOverflow(true) : null
```

Comment:
I think it us confusing to use ternaries when you use them for other purposes that assignments/value resolving.
I think this should be:
```typescript
if (isOverflowing) {
  setHasOverflow(true)
}

Reply:
Fair enough. I was coming from a place where I didn't want to nest if statements